src/ui/BlockEditor.tsx

```diff
@@ -280,68 +243,98 @@ export type Search = {
   setCM: (cm: CodeMirror.Editor) => void;
 };
 
-export type BlockEditorProps = typeof BlockEditor.defaultProps &
-  ConnectedProps<typeof blockEditorConnector> & {
-    value: string;
-    options?: Options;
-    cmOptions?: CodeMirror.EditorConfiguration;
-    keyMap?: { [index: string]: string };
-    /**
-     * id of the language being used
-     */
-    languageId: string;
-    search?: Search;
-    toolbarRef?: React.RefObject<HTMLInputElement>;
-    onBeforeChange?: IUnControlledCodeMirror["onBeforeChange"];
-    onMount: Function;
-    api?: API;
-    passedAST?: AST;
-    showDialog: Function;
-    ast: AST;
-  };
-
-class BlockEditor extends Component<BlockEditorProps> {
-  mouseUsed: boolean;
-  newAST: AST;
-  pendingTimeout: afterDOMUpdateHandle;
-
-  constructor(props: BlockEditorProps) {
-    super(props);
-    this.mouseUsed = false;
-
-    // stick the keyDown handler in the store
-    store.onKeyDown = this.handleKeyDown;
-
-    // NOTE(Emmanuel): we shouldn't have to dispatch this in the constructor
-    // just for tests to pass! Figure out how to reset the store manually
-    props.dispatch({ type: "RESET_STORE_FOR_TESTING" });
-  }
+export type BlockEditorProps = {
+  value: string;
+  options?: Options;
+  cmOptions?: CodeMirror.EditorConfiguration;
+  keyMap?: { [index: string]: string };
+  /**
+   * id of the language being used
+   */
+  languageId: string;
+  search?: Search;
+  toolbarRef?: React.RefObject<HTMLInputElement>;
+  onBeforeChange?: IUnControlledCodeMirror["onBeforeChange"];
+  onMount: Function;
+  api?: API;
+  passedAST?: AST; // not used?
+  showDialog: Function;
+};
 
-  static defaultProps = {
-    keyMap: defaultKeyMap,
-    search: {
+const BlockEditor = (props: BlockEditorProps) => {
+  const mounted = useRef(); // from https://stackoverflow.com/a/53406363/12026982
+  const dispatch: AppDispatch = useDispatch();
+  // grab some properties from state
+  const { cur, hasQuarantine } = useSelector((state: RootState) => {
+    console.log("@@@@@@@@@@@@@@@@", state);
+    return {
+      //ast: state.ast,
+      cur: state.cur,
+      hasQuarantine: !!state.quarantine,
+    };
+  });
+  // NOTE(Emmanuel): this was in matchDispatchToProps, but appears unused
+  const clearFocus = () => dispatch({ type: "SET_FOCUS", focusId: null });
+
+  const {
+    passedAST: ast,
```

Comment:
This is still different. In the old code, sometimes `ast` referred to `this.props.ast`, and other times it refers to `this.props.passedAST`, depending on what method you are in. Be very careful.